### DISCUSSION TITLE
Struct destructor

### DISCUSSION THREAD

**Discussion by sebas77** [dotnet/csharplang#5370] [Language Ideas] [answered]
Hi,

I found myself many times wanting a very simple ref struct destructor. It would be just a method that is called when the ref struct gets out of the scope. Nothing fancy, but I have use cases for it in my framework (I have never used disposing ref structs because I cannot trust the framework user to remember using the using keyword). Would it be technically possible to have it?

**Comment by koszeggy** [comment, marked as answer]
Duplicate of #1110

**Reply by sebas77** [reply]
thank you, copy and pasting this there, so you can delete this discussion

**Reply by redradist** [reply]
Not only, here is similar issue with design for destructor https://github.com/dotnet/csharplang/discussions/4808

From my disscussion with Microsoft folks they are very stubborn and need very specific arguments that only them knows which one ... I tired fighting with they, unfortunately I do not have impact on them ... (

**Reply by redradist** [reply]
Also Microsoft guys very opinionated ... It means that they do not try to look at features from other languages except Java and Kotlin and try to use them, for example: RAII, free functions, also do not try to add support at least for development on Linux (not for, only on Linux for Android and iOS) for MAUI Apps ...

... but I still hope that they will listen community and add at least this 3  these features !! ;)

**Comment by huoyaoyuan** [comment]
I think it's actually #4345
